src/main/java/org/opensearch/knn/quantization/quantizer/QuantizerHelper.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.quantizer;
+
+import org.opensearch.knn.quantization.models.requests.TrainingRequest;
+import lombok.experimental.UtilityClass;
+
+import java.util.BitSet;
+
+/**
+ * Utility class providing common methods for quantizer operations, such as parameter validation and
+ * extraction. This class is designed to be used with various quantizer implementations that require
+ * consistent handling of training requests and sampled indices.
+ */
+@UtilityClass
+class QuantizerHelper {
```

Comment:
Are we able to leverage lucene's VectorUtil class at all (https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/util/VectorUtil.java) so we can get SIMD performance benefits by default?

Reply:
I wanted to decopule framework as much as from Lucene , will take a second look on this  and if needed I can have this in future pr